ggml/src/llamafile/sgemm.cpp

```diff
@@ -985,6 +989,705 @@ class tinyBLAS_Q0_AVX {
 };
 #endif // __AVX__
 
+//PPC Implementation
+#if defined(__MMA__)
+
+#define SAVE_ACC(ACC, ii, jj) \
+   __builtin_mma_disassemble_acc(vec_C, ACC); \
+   for (int I = 0; I < 4; I++) { \
+      for (int J = 0; J < 4; J++) { \
+         *((float*)(C+ii+((jj+J)*ldc)+I)) = *((float*)&vec_C[I]+J); \
```

Comment:
It's probably better to do a 4 vector transpose here or invert the MMA inputs.  That way you can write vectors at a time instead of scalar elements.